tests/darwin/client_test.py

```diff
@@ -73,11 +74,53 @@ def test_returns_list_of_datasets(self, darwin_client: Client):
             assert_dataset(remote_datasets[0], expected_dataset_1)
             assert_dataset(remote_datasets[1], expected_dataset_2)
 
+    @responses.activate
```

Comment:
This is a socialised test, testing that the client is using the `_get_item_count` function correctly.

There is a functional unit test as well, testing the function in a pure functional way.